demos/browser/app/meetingV2/meetingV2.ts

```diff
@@ -1757,11 +1808,74 @@ export class DemoMeetingApp implements
       return DefaultDeviceController.synthesizeVideoDevice('smpte');
     }
 
-    if (value === 'None') {
-      return null;
+    return value;
+  }
+
+  private videoFilterToProcessor(videoFilter: string): VideoFrameProcessor | null {
+    this.log(`Choosing video filter ${this.videoFilter}`);
+
+    if (videoFilter === VideoFilterNameEnum.EMOJIFY) {
+      return new EmojifyVideoFrameProcessor('🚀');
     }
 
-    return value;
+    if (videoFilter === VideoFilterNameEnum.CIRCULAR_CUT) {
+      return new CircularCut();
+    }
+
+    if (videoFilter === VideoFilterNameEnum.NO_OP) {
+      return new NoOpVideoFrameProcessor();
+    }
+
+    return null;
+  }
+
+  private chosenDevice: { transformDevice: DefaultVideoTransformDevice | undefined; processor: VideoFrameProcessor;  innerDevice: Device} = {
+    transformDevice: undefined,
+    processor: undefined,
+    innerDevice: undefined
+  };
+
+  private async cleanUpChosenTransformDevice(): Promise<void> {
+    if (this.chosenDevice && this.chosenDevice.transformDevice) {
+      await this.chosenDevice.transformDevice.stop();
+    }
+    this.chosenDevice.transformDevice = null;
+    this.chosenDevice.processor = null;
+    this.chosenDevice.innerDevice = null;
+  }
+
+  private async videoInputSelectionWithOptionalFilter(intrinsicDevice: Device): Promise<VideoInputDevice> {
+    if (this.videoFilter === 'None') {
+      return intrinsicDevice;
+    }
+
+    if (this.chosenDevice.transformDevice && 
+      this.chosenDevice.processor && 
+      this.chosenDevice.processor.name() === this.videoFilter) {
+      // same processor, so just switching inner device
+      this.chosenDevice.transformDevice = this.chosenDevice.transformDevice.chooseNewInnerDevice(intrinsicDevice);
+      return this.chosenDevice.transformDevice;
+    }
+
+    const proc = this.videoFilterToProcessor(this.videoFilter);
+    // different processor
+    if (this.chosenDevice.transformDevice) {
+      await this.chosenDevice.transformDevice.stop();
+    }
+
+    const transformDevice = new DefaultVideoTransformDevice(new ConsoleLogger('Demo'), intrinsicDevice, [proc]);
```

Comment:
Reuse `this.logger`.

Reply:
we don't have a this.logger in DemoMeetingApp.

we have `MultiLogger`, `MeetingSessionPOSTLogger` created on demand, do you mind if I do another PR after this to clean that up a bit ?